src/api/wayfire/object.hpp

```diff
@@ -55,10 +58,13 @@ class signal_connection_t : public noncopyable_t
     void disconnect();
 
     class impl;
-    std::unique_ptr<impl> priv;
+    std::shared_ptr<impl> priv;
```

Comment:
This should remain unique_ptr .

Reply:
woops yeah that's leftover from my movable attempt. Basically unique_ptr still needs to know the sizeof() when destroying the object in some cases, but shared_ptr doesn't. Anyways, I'll remove this now.